test/ex_doc/autolink_test.exs

```diff
@@ -27,15 +27,28 @@ defmodule ExDoc.AutolinkTest do
     end
 
     test "private module" do
-      assert_unchanged("String.Unicode")
+      captured =
+        assert_warn(fn ->
+          assert_unchanged("String.Unicode")
+        end)
+
+      assert captured =~
+               "documentation references module \"String.Unicode\" but it is undefined or private"
     end
 
     test "erlang module" do
       assert_unchanged(":array")
     end
 
     test "unknown module" do
-      assert_unchanged("Unknown")
+      captured =
+        assert_warn(fn ->
+          assert_unchanged("Unknown")
+        end)
+
+      assert captured =~
+               "documentation references module \"Unknown\" but it is undefined or private"
```

Comment:
I don't think we should warn here, we want to write docs that have e.g. `` `MyApp` ``.

However, if we write `` [foo](`MyApp`) `` then we are referring to a particular module so if it doesn't exist, we should warn.

Reply:
so `String.Unicode` should warn, or not?

Reply:
To make it clear: linking to a module that has been defined with `@doc false` should warn or not?

Reply:
I think it should warn (and I initially implemented it as such) but I don't have a very strong opinion about it.

Reply:
done